editor/src/components/editor/store/dispatch.tsx

```diff
@@ -507,9 +512,33 @@ export function editorDispatch(
     storedState.workers.initWatchdogWorker(frozenEditorState.id)
   }
 
+  lastProjectContents = storedState.unpatchedEditor.projectContents
```

Comment:
I'm worried this function is becoming a spaghetti, could you please move this new block to a new function? (we should also break down the code above this into sub-functions)

Reply:
Yeah sure, I'll extract it. I wouldn't want it to live in `element-path.ts` since that pulls dependencies on the project model in with it, but it could certainly be tidied up here